### PR TITLE
Refactor filter code into standalone class.

### PR DESCRIPTION
**Pull request by jmichaelward**
Hey @sswebster - here's where things are at after working on this some more this evening. I've got 5 unit tests that I put into place before doing any refactoring, and then moved chunks around until I felt things were a little more logically grouped.

The main focus here is to have a single point of entry for the filtered data, and to leave the responsibility of the logs and other parts of getting the filtered data from the object to the add-on. I think this leaves things in a slightly better place for us to be able to change how things are structured (since there are now tests, we'll know when we're breaking some behavior), and better allows us to be able to test other parts of the separate class if needed, since the logic is now smaller (using Reflection, of course, because I've got everything except the main entry point to the object set to private).

Let's discuss tomorrow morning after my 1:1 with Steve.

### CODE REVIEW

File: includes/class-akismet-fields-filter.php [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+				// Normalize the label into a slug. See https://github.com/Automattic/jetpack/blob/43fee1286315992b343dd91601d5afad6c0f0d0f/modules/contact-form/grunion-contact-form.php#L2588
+				trim( // Strip all leading/trailing dashes.
+					preg_replace(   // Normalize everything to a-z0-9_-
+						'/[^a-z0-9_]+/',
+						'-',
+						strtolower( GFFormsModel::get_label( $field, 0, false, false ) ) // Lowercase
+					),
+					'-'
+				)
+			);
```
Comment (richardW8k) [inline includes/class-akismet-fields-filter.php:273]: seems like a good candidate for its own method

**Review by richardW8k — COMMENTED** [review]
Looking good. We'll probably want to drop "Enhancements" since the plan is to eventually move the entire integration here.

File: includes/class-akismet-fields-filter.php (hunk 1 above)
Comment (jmichaelward) [reply includes/class-akismet-fields-filter.php:273]: Absolutely!

**Comment by jmichaelward** [comment]
Merging this back into @sswebster's branch after talking through it together.

**Pull request merged**